x-pack/plugins/ml/public/application/components/anomalies_table/links_menu.tsx

```diff
@@ -265,12 +267,19 @@ export const LinksMenuUI = (props: LinksMenuProps) => {
 
       if (record.influencers) {
         kqlQuery = record.influencers
-          .map(
-            (influencer) =>
-              `"${influencer.influencer_field_name}":"${
-                influencer.influencer_field_values[0] ?? ''
-              }"`
-          )
+          .filter((influencer) => isDefined(influencer))
+          .map((influencer) => {
+            const fieldName = escapeQuotes(influencer.influencer_field_name);
```

Comment:
`fieldName` could be moved inside the `values.length > 0` check below, so it's only defined if it's going to be used.